source/particle/interpolator/quadratic_least_squares.cc

```diff
@@ -136,17 +187,61 @@ namespace aspect
                 A[9][particle_index] = relative_particle_position[2] * relative_particle_position[2];
               }
           }
+        std::vector<typename parallel::distributed::Triangulation<dim>::active_cell_iterator> active_neighbors;
+        GridTools::get_active_neighbors<parallel::distributed::Triangulation<dim>>(found_cell, active_neighbors);
+        for (const auto &active_neighbor : active_neighbors)
+          {
+            if (active_neighbor->is_artificial())
+              continue;
+            const std::vector<double> neighbor_cell_average = fallback_interpolator.properties_at_points(particle_handler, positions, selected_properties, active_neighbor)[0];
+            for (unsigned int property_index = 0; property_index < n_particle_properties; ++property_index)
+              {
+                if (selected_properties[property_index] == true && use_quadratic_least_squares_limiter[property_index] == true)
+                  {
+                    property_minimums[property_index] = std::min(property_minimums[property_index], neighbor_cell_average[property_index]);
+                    property_maximums[property_index] = std::max(property_maximums[property_index], neighbor_cell_average[property_index]);
+                  }
+              }
 
+          }
+        if (found_cell->at_boundary())
+          {
+            for (unsigned int face_id = 0; face_id < dim * 2; ++face_id)
+              {
+                if (found_cell->at_boundary(face_id))
+                  {
+                    // Faces 0 and 1, 2 and 3, and 4 and 5 oppose each other. Note by XOR with 1 we get the opposing face. In 2D faces 4 and 5 do not exist.
+                    const unsigned int opposing_face_id = face_id ^ 1;
+                    const auto &opposing_cell = found_cell->neighbor(opposing_face_id);
+                    if (!opposing_cell->is_locally_owned())
+                      {
+                        continue;
```

Comment:
At the moment yes, if we are not able to access the values in opposing_cell then we ignore it. This is definitely not the desired behavior, so if you have suggestions on this I am all ears. If I am not mistaken though if we neighbor a cell that is not locally owned, we are already in a ghost cell?
Since this limiter uses particles in neighboring cells, should I add this?
```AssertThrow(prm.get_bool("Update ghost particles") == true, ExcMessage("The quadratic least squares limiting scheme requires that 'Update ghost particles' is enabled."));```